noxfile.py

```diff
@@ -215,3 +236,63 @@ def _bump(session, *, version, file, kind):
     session.log(f"git commit")
     subprocess.run(["git", "add", str(file)])
     subprocess.run(["git", "commit", "-m", f"Bump for {kind}"])
+
+
+@contextlib.contextmanager
+def _replace_file(original_path):
+    # Create a temporary file
+    fh, replacement_path = tempfile.mkstemp()
+
+    try:
+        with os.fdopen(fh, "w") as replacement:
+            with open(original_path) as original:
+                yield original, replacement
+    except Exception:
+        raise
+    else:
+        shutil.copymode(original_path, replacement_path)
+        os.remove(original_path)
+        shutil.move(replacement_path, original_path)
+
+
+def _changelog_update_unreleased_title(version, *, file):
+    """Update an "*unreleased*" heading to "{version} - {date}"
+    """
+    yyyy_mm_dd = datetime.datetime.today().strftime("%Y-%m-%d")
+    title = f"{version} - {yyyy_mm_dd}"
+
+    with _replace_file(file) as (original, replacement):
+        for line in original:
+            if line == "*unreleased*\n":
+                replacement.write(f"{title}\n")
+                replacement.write(len(title) * "~" + "\n")
+                # Skip processing the next line (the heading underline for *unreleased*)
+                # since we already wrote the heading underline.
+                next(original)
+            else:
+                replacement.write(line)
+
+
+def _changelog_add_unreleased_title(*, file):
+    with _replace_file(file) as (original, replacement):
+        # Duplicate first 3 lines from the original file.
```

Comment:
What's being duplicated? Is it worth putting it in this file as a template instead to make sure this assumption doesn't change?

Reply:
Possibly. I'll explore in round 3.